src/config-loader/config-loader.ts

```diff
@@ -0,0 +1,293 @@
+import * as _express from 'express'
+
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+import * as path from 'path'
+import * as fs from 'fs'
+// We force the type to match the specific overloaded version we want, as promisify can only choose one :(
+const readFileAsync = Promise.promisify(fs.readFile) as any as (filename: string, encoding: string) => Promise<string>
+const readdirAsync = Promise.promisify(fs.readdir)
+
+import * as sbvrUtils from '../sbvr-api/sbvr-utils'
+const permissions = require('../sbvr-api/permissions')
+
+export interface SetupFunction {
+	(app: _express.Application, sbvrUtils: any, db: any, done: (err: any) => void): Promise<void> | void
```

Comment:
Shouldn't `done` be optional? Isn't this actually better as two overloaded functions -- one that returns void when given callback, another that doesn't accept a callback and returns promise